web-registry/src/pages/Marketplace/Marketplace.utils.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Convert Cosmos-style micro denom (starting with u) to natural denom.
+ * Example uregen -> REGEN
+ */
+export const formatDenomText = (askDenom: string): string => {
+  return askDenom.substring(1).toUpperCase();
+};
+
+/**
+ * Convert Cosmos-style micro denom amount to readable full-denom amount
+ * Example 7000000 uregen -> 7 REGEN
+ */
+export const microToDenom = (amount: string | number): number => {
+  const price = typeof amount === 'string' ? parseInt(amount) : amount;
+  return price / Math.pow(10, 6);
+};
+
+/**
+ * Convert readable full-denom amount to micro denom amount
+ * Example 8 REGEN -> 8000000 uregen
+ */
+export const denomToMicro = (amount: string | number): number => {
+  const price = typeof amount === 'string' ? parseInt(amount) : amount;
+  return price * Math.pow(10, 6);
+};
```

Comment:
Those are super useful functions 👍 
Should they be located outside of the marketplace though? Like in global utils.

Reply:
now lib/denom.utils